[Win] Skip CompileWebKitWithoutChange on Win-Tests-EWS

iangrunert opened the pull request:
#### 7ed5d8bb6c92ebf91ae3705bc86fcce6fb92a013
<pre>
[Win] Skip CompileWebKitWithoutChange on Win-Tests-EWS
<a href="https://bugs.webkit.org/show_bug.cgi?id=300919">https://bugs.webkit.org/show_bug.cgi?id=300919</a>

Reviewed by Jonathan Bedard.

It submits an auto-retry if compiling without the patch fails on that
box. Very occasionally saves a manual step for that PR at the expense
of delaying results for the other PRs waiting in the queue.

Given the limited worker resources I don&apos;t think that trade-off makes
sense for Win-Tests-EWS, we&apos;d rather free up the worker faster to work
on another job.

Canonical link: <a href="https://commits.webkit.org/305111@main">https://commits.webkit.org/305111@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb2edec51bfe578bbf58842f32209617b1d25648

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128752 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1009 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39582 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136134 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80131 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/130623 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/960 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/886 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98016 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/65934 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131699 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/718 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115337 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78624 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/656 "Passed tests") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [⏳ 🛠 wpe-cairo-libwebrtc ](https://ews-build.webkit.org/#/builders/WPE-Cairo-LibWebRTC-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109088 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33450 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138593 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/826 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/33944 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106565 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/128182 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/879 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/804 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106365 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28912 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/673 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111675 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53235 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/895 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30206 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/747 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/803 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/837 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4509,19 +4512,23 @@ def evaluateCommand(self, cmd):
-            ]
-            self.build.addStepsAfterCurrentStep(steps_to_add)
+
+            # The significant additional build time isn't worth it on Windows, we'd rather
```
It's ok if we don't care too much about this queue. But, for the record, the downside of this would be that this EWS queue will become extremely unreliable.

With this change, we won't be checking if the failures are pre-existing or not (except the basic check with results database). Any flaky/pre-existing failure on the tree can cause a red status-bubble for this queue on every subsequent PRs (even unrelated PRs).

Won't be a good user experience for the users, most users would have to learn to ignore when this EWS is red, also reducing the overall user confidence in EWS results.

iangrunert replied:
Yeah, it's a trade-off for sure.

At the moment, PRs are often merged before we get results for this queue; which makes it more likely that regressions are merged in the first place.

Happy to reverse this change if we find that it dramatically reduces the reliability, but I'm cautiously optimistic that this will be a positive change overall.

iangrunert committed:
[Win] Skip CompileWebKitWithoutChange on Win-Tests-EWS
https://bugs.webkit.org/show_bug.cgi?id=300919

Reviewed by Jonathan Bedard.

It submits an auto-retry if compiling without the patch fails on that
box. Very occasionally saves a manual step for that PR at the expense
of delaying results for the other PRs waiting in the queue.

Given the limited worker resources I don't think that trade-off makes
sense for Win-Tests-EWS, we'd rather free up the worker faster to work
on another job.

Canonical link: https://commits.webkit.org/305111@main
